config/ompi_check_ucx.m4

```diff
@@ -104,6 +104,13 @@ AC_DEFUN([OMPI_CHECK_UCX],[
                   old_CPPFLAGS="$CPPFLAGS"
                   AS_IF([test -n "$ompi_check_ucx_dir"],
                         [CPPFLAGS="$CPPFLAGS -I$ompi_check_ucx_dir/include"])
+                  # Turn off UCX version v1.8 due to issue #8321
+                  AC_PREPROC_IFELSE([AC_LANG_PROGRAM([#include <ucp/api/ucp_version.h>
+                                                      #if (UCP_API_MAJOR == 1) && (UCP_API_MINOR == 8)
+                                                      #error "Invalid version"
+                                                      #endif], [])],
+                                    [AC_MSG_RESULT([UCX version is ok])],
+                                    [AC_MSG_ERROR([UCX version 1.8 is unsupported due to known issue])])
```

Comment:
Minor nit -- can you make it (ignore my poor indenting here in github):

```
AC_MSG_CHECKING([UCX version])
AC_PREPROC_IFELSE([AC_LANG_PROGRAM([#include <ucp/api/ucp_version.h>
                                    #if (UCP_API_MAJOR == 1) && (UCP_API_MINOR == 8)
                                    #error "Invalid version"
                                    #endif], [])],
                                    [AC_MSG_RESULT([ok (not 1.8.x)])],
                                    [AC_MSG_RESULT([bad (1.8.x)])
                                     AC_MSG_ERROR([UCX version 1.8 is unsupported due to known issue])])
```

Reply:
Actually, wait -- this isn't quite right.

You don't want to AC_MSG_ERROR if you find a "bad" UCX, because it's possible that the user didn't ask for UCX at all.  If you find a "bad" UCX, you need to decide:

* If `--with-ucx` was used (i.e., if `$with_ucx` is not empty and is not `no`), then you should `AC_MSG_ERROR`, because the user explicitly asked for something that `configure` cannot do (because UCX is bad).
* If `--with-ucx` was not used (or if `--without-ucx` was used -- i.e., if `$with_ucx` is either empty or is `no`), then you can just disqualify building this component and let `configure` continue.  I.e., emit a warning, but do NOT invoke `AC_MSG_ERROR`.

Reply:
sure, will fix. BTW, is there some way to make autogen/configure run faster if I changed only ucx-related stuff?

Reply:
Try two things:

1. Set `AUTOMAKE_JOBS=x`, where X is a number of max threads/sub-processes that you will allow Automake to use (i.e., parallelism inside `automake`, which is run as part of `autogen.pl`).
2. Try using `./configure --cache-file=blah ...`, but I'm not 100% certain how well that will work if you are continually updating / re-running `autogen.pl` (i.e., you may get unexpected results).  It may not speed things up much, either.